### PR TITLE
Updated min tested Bazel version to 4.0.0

### PR DESCRIPTION
**Pull request by UebelAndre**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](https://github.com/bazelbuild/rules_python/blob/master/CONTRIBUTING.md) for info
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/bazelbuild/rules_python/issues/526

## What is the new behavior?

Updates the min tested Bazel version to `4.0.0`

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Other information
closes https://github.com/bazelbuild/rules_python/issues/526

### CODE REVIEW

**Review by hrfuller — APPROVED** [review]
I'm +1 on this, will wait for @alexeagle stamp as well. How risky is this upgrade, it may make future releases incompatible for people who can't go to newer bazel version right?

**Comment by alexeagle** [comment]
We'll need to document this as a breaking change, that users must upgrade to 4.0. We'll start using new features like the `json` module which will fail. It has been the LTS version for a while, and I think we are not required to support older ones, so policy-wise I believe this is fine. (We ought to have some testing against rolling releases as well. I hope the rules authors SIG will make some guidance about what rulesets are expected to do)

One thing to figure out (maybe before merging this?) is how to give the right error message so developers know they must upgrade, instead of just reporting [name 'json' is not defined](https://github.com/bazelbuild/rules_nodejs/issues/2938)
@UebelAndre could you find the right spot to hook up a `versions.check` https://docs.aspect.dev/bazelbuild/bazel-skylib/1.0.3/docs/versions.html#versions-check
?

**Comment by UebelAndre** [comment]
I unfortunately don't have a whole lot of bandwidth so don't think I'll be able to get to that in the near future 😞 Is this something I can leave to the `rules_python` maintainers?

**Comment by alexeagle** [comment]
I think I could find time for it. I'm looking around for where the check could go - there's no single call from users WORKSPACE files into a repository rule that we could put it. So I guess we just check at the beginning of several of them like `pip_install` and `pip_import`

**Comment by alexeagle** [comment]
@googlebot I consent.

File: python/pip_install/repositories.bzl
Comment (hrfuller) [inline python/pip_install/repositories.bzl:6]: Looks like versions.bzl tries to load skylib anyway. https://github.com/bazelbuild/rules_python/pull/533/files#diff-86c5bb17276af4fdb85612635eabfdaf9b91676ae6bac37ee78a09947dfd6045R1

**Comment by hrfuller** [comment]
@googlebot I consent.

File: python/pip_install/repositories.bzl
Comment (alexeagle) [reply python/pip_install/repositories.bzl:6]: thanks for fixing it!

**Pull request merged**
